docs/development/getting-started-locally.md

```diff
@@ -0,0 +1,162 @@
+
+## Clone the etcd-druid github repo
+
+```sh
+# clone the repo
+git clone https://github.com/gardener/etcd-druid.git
+# cd into etcd-druid folder
+cd etcd-druid
+```
+
+---
+
+> **Note:**
+>
+>- Etcd-druid uses [kind](https://kind.sigs.k8s.io/) as it's local Kubernetes engine. The local setup is configured for kind due to it's convenience but any other kubernetes setup would also work.
+>- To setup Etcd-Druid with backups enabled on a [LocalStack](https://github.com/localstack/localstack) provider, refer [this document](getting-started-locally-localstack.md)
+>- In the section [Annotate Etcd CR with the reconcile annotation](#annotate-etcd-cr-with-the-reconcile-annotation), the flag `ignore-operation-annotation` is set to false, which means manual annotation to the Etcd CR is requied to reconcile the Etcd CR. To disable this and auto reconcile the Etcd CR for any change in the Etcd yaml, set the`ignoreOperationAnnotation` flag to `true` in the `values.yaml` located at `$PWD/charts/druid/values.yaml`. Or if the Etcd-Druid is being run as a process, then while starting the process, attach the CLI flag `--ignore-operation-annotation=true` to it.
+
+## Setting up the kind cluster
+
+```sh
+# Create a kind cluster
+make kind-up
+```
+
+This creates a new kind cluster and stores the kubeconfig in the  `./hack/e2e-test/infrastructure/kind/kubeconfig` file.
+
+To target this newly created cluster, set the `KUBECONFIG` environment variable to the kubeconfig file located at `./hack/e2e-test/infrastructure/kind/kubeconfig` by using the following
+
+```sh
+export KUBECONFIG=$PWD/hack/e2e-test/infrastructure/kind/kubeconfig
+```
+
+## Setting up etcd-druid
+
+```sh
+make deploy
+```
+
+This generates the `Etcd` CRD and deploys an etcd-druid pod into the cluster
+
+### Prepare the Etcd CR
+
+Etcd CR can be configured in 2 ways. Either to take backups to the store or disable them. Follow the appropriate section below based on the requirement.
+
+The Etcd CR can be found at this location `$PWD/config/samples/druid_v1alpha1_etcd.yaml`
+
+- **Without Backups enabled**
+
+    To setup Etcd-druid without backups enabled, make sure the `spec.backup.store` section of the Etcd CR is commented out.
+
+- **With Backups enabled (On Cloud Provider Object Stores)**
+
+  - **Prepare the secret**
+
+    Create a secret for cloud provider access. Find the secret yaml templates for different cloud providers [here](https://github.com/gardener/etcd-backup-restore/tree/master/example/storage-provider-secrets).
+
+    Replace the dummy values with the actual configurations and make sure to add a name and a namespace to the secret as intended.
+
+    > **Note 1:** The secret should be applied in the same namespace as druid.
+    >
+    > **Note 2:** All the values in the data field of secret yaml should be in base64 encoded format.
+
+  - **Apply the secret**
+
+    ```sh
+    kubectl apply -f path/to/secret
+    ```
+
+  - **Update Etcd**
```

Comment:
```suggestion
  - **Adapt `Etcd` resource**
```

The term `update` can be misread as "update the etcd resource using kubectl", rather than "uncomment the section within the etcd manifest file". Hence, you can probably use `Adapt resource` as a more unambiguous word